parsec/core/gui/device_recovery_export_widget.py

```diff
@@ -82,32 +94,38 @@ def __init__(self, devices, parent=None):
                 f"{device.organization_id.str} - {device.user_display}", device.slug
             )
 
-    def _on_select_file_clicked(self):
+    def _on_select_file_clicked(self) -> None:
         key_dir = QDialogInProcess.getExistingDirectory(
             self, translate("TEXT_EXPORT_KEY"), str(Path.home())
         )
         if key_dir:
             self.label_file_path.setText(key_dir)
         self._check_infos()
 
-    def _check_infos(self):
+    def _check_infos(self) -> None:
         self.info_filled.emit(len(self.label_file_path.text()) > 0)
 
-    def get_selected_device(self):
+    def get_selected_device(self) -> LocalDevice | AvailableDevice:
         return self.devices[self.combo_devices.currentData()]
 
-    def get_save_path(self):
+    def get_save_path(self) -> PurePath:
         return PurePath(self.label_file_path.text())
 
 
 class DeviceRecoveryExportWidget(QWidget, Ui_DeviceRecoveryExportWidget):
     export_success = pyqtSignal(QtToTrioJob)
     export_failure = pyqtSignal(QtToTrioJob)
 
-    def __init__(self, config, jobs_ctx, devices, parent=None):
+    def __init__(
+        self,
+        config: CoreConfig,
+        jobs_ctx: QtToTrioJobScheduler,
+        devices: Sequence[LocalDevice | AvailableDevice],
```

Comment:
Same thing as before, should change the code so that only `AvailableDevice` are used here.